### ISSUE TITLE
Device Disappearing From Other Devices After A While

### ISSUE THREAD

**Issue by iSteve-O** [NorthernMan54/homebridge-yamaha-zone-tv#29]
I am having an issue where the device will appear fine on all devices, and then in the home app on my iPhone it will be fine but does not appear in my home on the iPad or MacBook after a while. I was able to fix this before by deleting the device and adding it again but this is not convenient because all scenes and automations need to be redone afterward.

I hope there's a fix for this. Thanks for all your great work on the plugin. I don't expect an update because it's been so long, but I felt like I had to report the issue. 

Have a great day!

**Comment by NorthernMan54** [comment]
To be honest I have no clue what would cause this. Is this your only Homebridge device ?

Is it possibly a flavour of this ?

https://github.com/NorthernMan54/homebridge-yamaha-home/issues/41

**Comment by iSteve-O** [comment]
Yeah it is a strange issue for sure. I have many devices in homebridge, including "TVs" through this plugin, 1 Bravia TV & 2 through Broadlink ("Cable Box" +Speaker & "Yamaha IR" +Speaker -screenshot below). 

The only one that ever does this is the device on this Yamaha-zone plugin, but I can see it just fine on my Macs, iPhone & iPad, and it works perfectly on all of them right after adding it, and then after a while it will just not be there when I go to open the home app on one or a few of these apple devices, but never all of them; it's always on at least one of them working perfectly. 

The device always seems to be there on my iPhone, but I've had it disappear from both my iPad & Macs in the past. Right now I can see/use it on both my iPad, iPhone, & my 2015 MBP, but not on my 2016 MBP. I attached screenshots from just now showing all 4 device's idea of what's in the "Living Room".

It's working perfectly other than that, so I know it has to be configured correctly. I'm aware it has to be on when HB restarts and all of the quirks that may happen with these things, but this has me baffled. I hope your genius allows you to figure this out because it really works beautifully otherwise. if you look at the images below, you'll see it's just the only thing missing from my 1 Mac ("Yamaha").

I appreciate your attention to the matter, as I was not expecting a reply!

**Fine on iPhone:**
![FineOniPhone](https://user-images.githubusercontent.com/10133108/158522758-2ebf956f-19a6-4fa7-9c3d-b38ef8d06265.jpeg)

**Fine on iPad:**
![FineOniPad](https://user-images.githubusercontent.com/10133108/158522768-e33b6969-d340-4326-8b2c-0df1984594a9.jpeg)

**Fine on 2015 MacBook Pro:**
![FineOnMac1](https://user-images.githubusercontent.com/10133108/158522778-01d21ef7-4a23-48d6-b3f9-d8b29adcecd9.png)

**Absent on 2016 MacBook Pro:**
![NotOnMac2](https://user-images.githubusercontent.com/10133108/158522785-1f108ba4-8947-47fd-ac5b-a0427df51248.jpg)

**Homebridge Accessories:**
![HBAccessories](https://user-images.githubusercontent.com/10133108/158523345-78a11e93-4946-4622-8e32-e9a2e1b81cae.png)

**Comment by NorthernMan54** [comment]
Are there any error messages logged by Homebridge ?

What I'm hypothesizing is a minor issue with the plugin's definition of this device or a data value from the device that causes the Home App on your 2016 MacBook Pro to not display the device.  The Home app is pretty rigid around what services a device can have, and does occasionally behave oddly when an unexpected value is received.

The trick is troubleshooting this, as the Home app does not have a visible error log that can be scanned for errors or bad data ( The Console app will show logs from the process 'homed', but I no clue what it is saying )

**Comment by iSteve-O** [comment]
I just noticed that in the images above even though the device is showing on the 2015 MBP Home app it’s not showing up in the scenes because all the input scenes appear active (cable, Apple TV, PlayStation). That’s even stranger. 

So, with no debug logging enabled, no, I’m getting no errors in the log at all, but they’d only be major ones. Can I enable debug logging for this plugin only, or do I need to enable all logging for HB to see if there’s any error?

I do understand about the difficulty with apple logs. 

I really appreciate your help!

**Comment by NorthernMan54** [comment]
I was thinking about these type of log messages, which are just logged, no additional settings or debug required.

```
"This plugin generated a warning from the characteristic 'Target Heating Cooling State': characteristic value 3 is not contained in valid values array. See https://git.io/JtMGR for more info."
```

If these were coming out, it would be an easy fix

**Comment by iSteve-O** [comment]
No nothing at all like that. I just put HB in debug mode and restarted and I only get a few things that all look proper. I just changed inputs on the receiver while logging in HB also and no messages at all. all I get in the log is this...

`Publishing bridge accessory (name: Yamaha, publishInfo: {
  username: '0E:48:7D:4C:DF:27',
  port: 46640,
  pincode: '***-**-***',
  category: 2,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'ciao'
...
  Publishing external accessory (name: Yamaha, publishInfo: {
  username: 'BB:23:AE:CD:28:DB',
  pincode: '***-**-***',
  category: 34,
  port: undefined,
  bind: [ 'eth0', [length]: 1 ],
  mdns: undefined,
  addIdentifyingMaterial: true,
  advertiser: 'ciao'`

**Comment by zenfty** [comment]
this still happens
